internal/client/client.go

```diff
@@ -180,13 +185,14 @@ func (c *Client) Shutdown(ctx context.Context) error {
 		ctx = context.Background()
 	}
 	c.EventDispatcher.Publish(lifecycle.NewLifecycleStateChanged(lifecycle.StateShuttingDown))
+	c.ExecuteShutdownHandlers(ctx)
```

Comment:
Shutdown handlers are called twice, here and below. Why is that required?

Reply:
Upcoming loop was redundant. Thanks for that catch. Removed it. [a6b3aef](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/a6b3aef218b08ee2bee4501ee356f35c96a4e97e).